### PR TITLE
Feature/dynamic backgrounds

### PR DESCRIPTION
**Pull request by lisalootens**

### CODE REVIEW

File: src/components/backgrounds/Background.tsx [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
+  currentMonth === 12 && currentDay > 6 && currentDay < 13;
 
-export const Background = () =>
-  isBirthdayBackgroundInRange ? <BirthdayBackground /> : <CirclesBackground />;
+const isSnowBackgroundInRange =
+  // When birthday background is not active, show a snowy background during winter time
+  isBirthdayBackgroundInRange ? false : currentMonth >= 11 || currentMonth <= 2;
+
+export function Background() {
+  switch (true) {
```
Comment (jobveldhuis) [inline src/components/backgrounds/Background.tsx:18]: Please don't switch over true - it's a code smell. In this case, it would be best to:

```tsx
export function Background() {
  if (isBirthdayBackgroundInRange) return <BirthdayBackground />;
  if (isSnowBackgroundInRange) return <SnowBackground />;
  return <CirclesBackground />
```

In this case, you wouldn't even need to check whether the birthday background is in range for the snow background, because of the pattern of an early exit when this is the case.

File: src/components/backgrounds/Background.tsx [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
 
-// Every year between the first and 13th of December, show a birthday background
 const isBirthdayBackgroundInRange =
-  currentMonth === 12 && currentDay >= 1 && currentDay < 13;
+  // Every year between the first and 13th of December, show a birthday background
+  currentMonth === 12 && currentDay > 6 && currentDay < 13;
 
-export const Background = () =>
-  isBirthdayBackgroundInRange ? <BirthdayBackground /> : <CirclesBackground />;
+const isSnowBackgroundInRange =
```
Comment (jobveldhuis) [inline src/components/backgrounds/Background.tsx:13]: Semantically, you would prefer to make this a boolean comparison, not a ternary operation - since this really isn't a choice between two. Instead, it's a scenario of (A and B), leading to the following:

```ts
const isSnowBackgroundInRange = !isBirthdayBackgroundInRange && (currentMonth >= 11 || currrentMonth <= 2);
```

Also, depending on the structure of the code that renders, you might not need this initial check before the `&&`. See my comment below.

Comment (jobveldhuis) [inline src/components/backgrounds/Background.tsx:13]: Naming variables is hard, but I would say you should reconsider naming these two variables. For example, something like:

```ts
const isBirthdayClose = true;
const isWinter = true;
```

Which would more clearly convey the meaning of the right hand side of these assignments.

File: src/components/backgrounds/SnowBackground.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React, { JSX } from "react";
+import "./animations.css";
+
+function getRandomInt(min: number, max: number) {
```
Comment (jobveldhuis) [inline src/components/backgrounds/SnowBackground.tsx:4]: What does `getRandomInt` return? Of course, we know this returns a random integer, but why not type it this way as well. This would help the compiler step through your code more quickly, also it will catch any unwanted bugs in the future if you're a bit more specific here.

File: src/components/backgrounds/SnowBackground.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  return Math.random() * (max - min) + min;
+}
+
+export function SnowBackground(): JSX.Element {
+  return (
+    <div className="absolute z-[-1] opacity-80">
+      <ul>
+        {Array.from({ length: 200 }).map((_, index) => {
+          // Generate random values within specified ranges
```
Comment (jobveldhuis) [inline src/components/backgrounds/SnowBackground.tsx:13]: This should probably be a component, you're doing a lot of calculations here - which should only exist in the component. For example a `SnowFlake` component, or whatever. This could internally calculate and then render to this screen.

**Review by jobveldhuis — CHANGES_REQUESTED** [review]

File: src/components/backgrounds/SnowBackground.tsx (hunk 3 above)
Comment (lisalootens) [reply src/components/backgrounds/SnowBackground.tsx:4]: Sorry, I don't quite get what you mean.

File: src/components/backgrounds/SnowBackground.tsx (hunk 4 above)
Comment (lisalootens) [reply src/components/backgrounds/SnowBackground.tsx:13]: Agreed - lots of calculations and this might not really fit here, but don't you want to keep these sort of things within the component? Does it really need to be a separate component?

File: src/components/backgrounds/SnowBackground.tsx (hunk 3 above)
Comment (jobveldhuis) [reply src/components/backgrounds/SnowBackground.tsx:4]: If you add a return type, you can help the compiler and type checker a bit with these kind of functions. It's good practice to not always do this, because sometimes you *want* to infer the return type based on the function return statements, but when you don't and you are sure what it should return, it's good practice to add a return type.

```ts
function getRandomInt(min: number, max: number): number {
    ...
}
```

Say you update this function without adding a return type:
```ts
// Stupid code for demonstration purposes only
function getRandomInt(min: number, max: number) {
  return Math.random() > 0.5 ? 1 : "Test"
}
```

This would be perfectly valid and all your code that depends on it now is "broken". For everywhere you call this function, the variable attached will be of type `string | number` and in bigger code bases this may lead to an absolute nightmare of recovery work if you don't identify the source of this issue quickly. If you add a return type, however, and you do the same change - the TS compiler will error out on *this function and this function only*, so you have identified the source without having to do anything.

TL:DR - Add return types when you know what functions should return and you don't want to infer a return type based on returns in the function body.

File: src/components/backgrounds/SnowBackground.tsx (hunk 4 above)
Comment (jobveldhuis) [reply src/components/backgrounds/SnowBackground.tsx:13]: The way I see it, components are functions and functions should do one thing and one thing only. In this case, you want to have a component that renders the wrapper around the snowflakes and you want to have the snowflakes themselves. These are two separate components, because they do two vastly different things. Where you add this component is up to you: either in a separate file or in this file. I try to stick to the "one component per file" rule where possible, because it makes codebases more readable.
